frontend/ui/rendertext.lua

```diff
@@ -233,4 +233,22 @@ function RenderText:renderUtf8Text(dest_bb, x, baseline, face, text, kerning, bo
     return pen_x
 end
 
+local ellipsis, space = "…", " "
+local ellipsis_width, space_width
+function RenderText:truncateTextByWidth(text, face, max_width, prepend_space)
+    if not ellipsis_width then
+        ellipsis_width = RenderText:sizeUtf8Text(0, max_width, face, ellipsis).x
```

Comment:
We are already in the "class" method, so we can now replace all `ReaderText` with `self`.